model/riscv_insts_zicsr.sail

```diff
@@ -72,7 +72,12 @@ function readCSR csr : csreg -> xlenbits = {
     (0xC22, _) => vlenb,
 
     /* trigger/debug */
-    (0x7a0,  _) => ~(tselect),  /* this indicates we don't have any trigger support */
+    (0x7a0,  _) => tselect.bits,
```

Comment:
This is missing legalisation functions.

Reply:
It is in the read_CSR function. does this even need legalization?

Reply:
In the RISCV model, we do write legalisation today, but I don't think this is sufficient for the Sdtrig spec.

Reply:
Actually, I am new to Sail and sail-riscv and still learning about many things. Any help is appreciated. Also, this is not complete as I am first completing the registers and will be working on the legalization after they are perfectly integrated and the repo is able to compile. Please, give your suggestions in this regard if you find some bug or something helpful.

Reply:
Sdtrig is a rather complex specification, for example because the legalisation of some registers depends on the values of some other registers. Sdext is much easier

Reply:
Yes I understand, but this is assigned to me as everyone else who was working on it left the team due to other priorities and this is a priority according to [this.](https://github.com/orgs/riscv-admin/projects/2/views/19?pane=issue&itemId=23160706)

Reply:
> In the RISCV model, we do write legalisation today, but I don't think this is sufficient for the Sdtrig spec. But maybe if you only implement the cases where `action=0` and `dmode=0` this may be enough

According to this [comment](https://github.com/riscv-admin/dev-partners/issues/8#issuecomment-2099470840) in the conversation, I think that this should be enough. What do you think?

Reply:
If `action=0` and `dmode=0` then you are not doing any triggering, and write legalisation is probably enough. However none of the hard cases in the verification of the debug extensions will become visible in this way.